redins.py

```diff
@@ -1,244 +1,185 @@
+            # print("getting long lived access token")
+            r = __requests.get(
+                url="https://graph.facebook.com/v5.0/oauth/access_token",
+                params={
+                    "grant_type": "fb_exchange_token",
+                    "client_id": app_id,
+                    "client_secret": app_secret,
+                    "fb_exchange_token": user_access_token
+                })
+            llt = r.json()["access_token"]
+            print("Long lived access token is :", llt)
+            return llt
+        except KeyError:
+            print("unable to get long lived access token")
 
-    for j in range(num):
+    def get_page_access_token(self, app_id, long_lived_token):
         try:
-            files = __random.choice(dirs)
-            files = __os.path.join(curr_dir, 'red_media', files)
-            tweet.PostUpdate(__random.choice(caps), files)
-            print("Uploaded..")
-            __os.remove(files)
-            __time.sleep(10)
+            # print("Getting Page access Token")
+            URL = "https://graph.facebook.com/v5.0/"+app_id
+            r = __requests.get(
+                url=URL,
+                params={
+                    "fields": "access_token",
+                    "access_token": long_lived_token
+                })
+            # print(r.json())
+            page_access_token = r.json()["access_token"]
+            return page_access_token
+        except KeyError:
+            print("Invalid page access token."
+                  "Please check your user access tokens")
+
+    def login(self):
+        if self.twitter:
+            self.twitter_client = twitter.Api(**self.config.creds["twitter"])
+        if self.instagram:
+            insta_creds = self.config.creds["instagram"]
+            self.instagram_client = __ig(
+                insta_creds["username"], insta_creds["password"])
+            self.instagram_client.login()
+        if self.facebook:
+            facebook_creds = self.config.creds["facebook"]
+            page_access_token = facebook_creds["page_access_token"]
+            user_access_token = facebook_creds["user_access_token"]
+            app_id = facebook_creds["app_id"]
+            app_secret = facebook_creds["app_secret"]
+            if self.check_access_token(page_access_token):
+                self.facebook_client = facebook.GraphAPI(
+                    access_token=page_access_token,
+                    version="3.0")
+            elif self.check_access_token(user_access_token):
+                long_lived_token = self.get_long_access_token(
+                    app_id, app_secret, user_access_token)
+                page_access_token = self.get_page_access_token(
+                    app_id, long_lived_token)
+                self.facebook_client = facebook.GraphAPI(
+                    access_token=page_access_token,
+                    version="3.0")
+            else:
+                print("Invalid Tokens Found !!!")

```

Comment:
> print("Invalid Tokens Found !!!")

Change this to 
print("Invalid token found for Facebook. Will not post on Facebook.")